pkgs/development/haskell-modules/configuration-hackage2nix.yaml

```diff
@@ -604,7 +604,7 @@ default-package-overrides:
   - deriving-compat ==0.5.7
   - derulo ==1.0.5
   - detour-via-sci ==1.0.0
-  - dhall ==1.24.0
+  - dhall ==1.27.0
```

Comment:
This list is `default-package-overrides`.  At the top, you can see a comment explaining that these packages are from LTS-14:

https://github.com/NixOS/nixpkgs/blob/fb5bfb25853f91d5b980d3db1ff218d6cfc974cd/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L46

In general, nixpkgs tracks the latest LTS release, and we don't override the versions of packages in the LTS release (since that would somewhat defeat the purpose of tracking the LTS release).

Why do you want a newer version of `dhall` in nixpkgs?

If this is just for your own use, maybe you'd be fine with using something like `callHackage` or `callCabal2nix` to get the latest version of `dhall`?

https://github.com/NixOS/nixpkgs/blob/fb5bfb25853f91d5b980d3db1ff218d6cfc974cd/pkgs/development/haskell-modules/make-package-set.nix#L178-L182

https://github.com/NixOS/nixpkgs/blob/fb5bfb25853f91d5b980d3db1ff218d6cfc974cd/pkgs/development/haskell-modules/make-package-set.nix#L213

-------------------------------------------------------------------------------------

Also, for future reference these types of changes generally go to the `haskell-updates` branch, not `master`.

Reply:
I want to update https://github.com/NixOS/nixpkgs/blob/92a047a6c4d46a222e9c323ea85882d0a7a13af8/pkgs/development/interpreters/dhall/default.nix

Changing
```
  static = haskell.lib.justStaticExecutables haskellPackages.dhall;
```

To

```
  static = haskell.lib.justStaticExecutables haskellPackages.dhall_1_27_0;
```

Works, but I don't think this is good idea

What should I do?

Reply:
@srghma I think using `dhall_1_27_0` is the best bet here if you want the latest version of `dhall`.  It is annoying that you'd have to update it manually every time a new version of `dhall` comes out, but that's really no different than any other package in nixpkgs.

It doesn't look like `pkgs/development/interpreters/dhall/default.nix` has a maintainer, so I'm not sure who would be a good person to contact about this.

It looks like @Profpatsch is the last person to edit this file, so I'd be interested in their opinion.

@srghma I'd suggest you change this PR to instead make `pkgs/development/interpreters/dhall/default.nix` explicitly use `dhall_1_27_0`, and then see if @Profpatsch has any suggestions.

If we don't hear anything from @Profpatsch (or anyone else) for a week or so, I'll merge it in.

Reply:
ok

Reply:
done